vpr/src/base/vpr_context.h

```diff
@@ -369,6 +370,31 @@ struct RoutingContext : public Context {
         cached_router_lookahead_;
 };
 
+/**
+ * @brief State relating to VPR's floorplanning constraints
+ *
+ * This should contain only data structures related to constraining blocks
+ * to certain regions on the chip.
+ */
+struct FloorplanningContext : public Context {
```

Comment:
mention in comments -> constraints are input into program, don't change. clustering_constraints will change if clusters change (basically talking about lifetime of the two variables)

Reply:
cluster_constraints computed during clustering